pkg/operation/botanist/health_check.go

```diff
@@ -265,6 +265,12 @@ func computeRequiredControlPlaneDeployments(
 		}
 	}
 
+	if gardencorev1beta1helper.ShootWantsVerticalPodAutoscaler(shoot) {
```

Comment:
This will actually cause failing healthcheck for `< 24h` for ShootedSeeds with `.spec.confineSpecUpdateRollout=true` that now set `.spec.kubernetes.verticalPodAutoscaler.enabled=true`, right?

Reply:
Nice catch! ❤️  Indeed, we should probably check here whether there is a `Deployment` for e.g. the `vpa-admission-controller` at all before adding the component names to the required names (yes, this makes the "deployment exists" check somewhat obsolete).

Reply:
I think we have the same problem for the cluster autoscaler, haven't we?

Reply:
Yep, you are right. The same applies also for the cluster-autoscaler deployment health check.

Reply:
This health check could have been executed only if `generation == observedGeneration`, but with #2233 the `generation` is not bumped when the spec is changed, thus this comparison cannot be used.

The other option is to extend the shoot status with a field that reflects that actual deployment status of the VPA.

Reply:
Would it be an option to leverage the shoot spec of the `Cluster` resource to determine whether to perform these checks or not?

Reply:
Very good idea.

Reply:
Yep, nice idea 👍

Reply:
After having another look at the code, I've spotted more areas where we need to rely on the cluster resource. I tend to open a separate and dedicated PR if you don't mind?

Reply:
Okay, I think I am fine as it is currently. Thank you for checking it.